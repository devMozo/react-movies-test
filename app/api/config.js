export const URL_MOVIE_API = 'https://cors-anywhere.herokuapp.com/https://api.themoviedb.org/3';
