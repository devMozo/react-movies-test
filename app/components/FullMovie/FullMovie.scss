.FullMovie {
    .FullMovie__header {
        width: 100%;
        background-color: #111;
        text-align: left;
        color: white;
        padding: 40px;
        box-sizing: border-box;

        & > * {
            display: inline-block;
            vertical-align: middle;
        }

        img {
            width: 200px;
            height: 200px;
            object-fit: cover;
            margin-right: 40px;
            box-shadow: 0 0 10px #000;
        }
    }

    h3 {
        color: #333;
        font-size: 20px;
        padding: 20px 0 0;
    }

    .FullMovie__description {
        background-color: #eee;
        color: #333;
        padding: 40px;
        box-sizing: border-box;
    }

    .FullMovie__features {
        padding: 0;
        margin: 40px 0 60px;

        li {
            list-style: none;
            color: #333;

            b {
                color: #aaa;
            }
        }
    }
}
