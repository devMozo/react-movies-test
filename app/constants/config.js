export const MOVIES_API_KEY = '1038f4f1dddddb86a86db5baf8cfaf04';
export const MOVIES_IMG_PATH = 'https://image.tmdb.org/t/p/w500/';
