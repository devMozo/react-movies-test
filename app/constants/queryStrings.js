export const QUERY_API_KEY = 'api_key';
