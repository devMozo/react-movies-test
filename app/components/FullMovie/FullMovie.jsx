import React from 'react';
import { MOVIES_IMG_PATH } from '../../constants/config';
import PropTypes from 'prop-types';
import './FullMovie.scss';

const FullMovie = ({ movie }) => {
    const {
        poster_path,
        vote_average,
        title,
        overview,
        production_companies,
        production_countries,
        status,
        budget,
        adult,
        release_date,
    } = movie;

    return (
        <section className="FullMovie">
            <header className="FullMovie__header">
                <picture>
                    <img src={`${MOVIES_IMG_PATH}${poster_path}`} />
                </picture>
                <section className="FullMovie__info">
                    <h2> {title} </h2>
                    <h4> Rating: {vote_average} </h4>
                </section>
            </header>
            <section className="FullMovie__description">
                <p> {overview} </p>
            </section>
            <h3> Features </h3>
            <ul className="FullMovie__features">
                <li>
                    {' '}
                    <b>Production Companies:</b>{' '}
                    {production_companies && production_companies.map(item => item.name).join(', ')}{' '}
                </li>
                <li>
                    {' '}
                    <b>Production Countries:</b>{' '}
                    {production_countries && production_countries.map(item => item.name).join(', ')}{' '}
                </li>
                <li>
                    {' '}
                    <b>Status:</b> {status}{' '}
                </li>
                <li>
                    {' '}
                    <b>Budget:</b> {budget}{' '}
                </li>
                <li>
                    {' '}
                    <b>Only Adult:</b> {adult ? 'Yes' : 'No'}{' '}
                </li>
                <li>
                    {' '}
                    <b>Release Date:</b> {release_date}{' '}
                </li>
            </ul>
        </section>
    );
};

FullMovie.propTypes = {
    movie: PropTypes.object,
};

export default React.memo(FullMovie);
