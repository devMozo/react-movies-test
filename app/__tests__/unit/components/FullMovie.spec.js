import React from 'react';
import { shallow } from 'enzyme';
import FullMovie from '../../../components/FullMovie/FullMovie';
import { MOVIES_IMG_PATH } from '../../../constants/config';

describe('<FullMovie />', () => {
    const PROPS = {
        movie: {
            poster_path: 'mocked_poster',
            vote_average: 2,
            title: 'mocked_title',
            overview: 'mocked_overview',
            production_companies: [
                {
                    name: 'Company 1',
                },
                {
                    name: 'Company 2',
                },
            ],
            production_countries: [
                {
                    name: 'Country 1',
                },
                {
                    name: 'Country 2',
                },
            ],
            status: 'Released',
            budget: 123456789,
            adult: false,
            release_date: '17-2-2019',
        },
    };
    let component;

    beforeEach(() => {
        component = shallow(<FullMovie {...PROPS} />);
    });

    it('should render correctly in "debug" mode', () => {
        expect(component).toMatchSnapshot();
    });

    it('should render correctly each field', () => {
        const image = component.find('.FullMovie__header img');
        const title = component.find('.FullMovie__info h2');
        const overview = component.find('.FullMovie__description p');
        const features = component.find('.FullMovie__features li');

        expect(image.prop('src')).toBe(`${MOVIES_IMG_PATH}${PROPS.movie.poster_path}`);
        expect(title.contains(PROPS.movie.title)).toBeTruthy();
        expect(overview.contains(PROPS.movie.overview)).toBeTruthy();
        expect(features.length).toBe(6);
    });
});
