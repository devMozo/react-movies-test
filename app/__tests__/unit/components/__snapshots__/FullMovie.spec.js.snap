// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullMovie /> should render correctly in "debug" mode 1`] = `
<section
  className="FullMovie"
>
  <header
    className="FullMovie__header"
  >
    <picture>
      <img
        src="https://image.tmdb.org/t/p/w500/mocked_poster"
      />
    </picture>
    <section
      className="FullMovie__info"
    >
      <h2>
         
        mocked_title
         
      </h2>
      <h4>
         Rating: 
        2
         
      </h4>
      <button
        className="FullMovie__add-button"
      >
        Buy Item
      </button>
    </section>
  </header>
  <section
    className="FullMovie__description"
  >
    <p>
       
      mocked_overview
       
    </p>
  </section>
  <h3>
     Features 
  </h3>
  <ul
    className="FullMovie__features"
  >
    <li>
       
      <b>
        Production Companies:
      </b>
       
      Company 1, Company 2
       
    </li>
    <li>
       
      <b>
        Production Countries:
      </b>
       
      Country 1, Country 2
       
    </li>
    <li>
       
      <b>
        Status:
      </b>
       
      Released
       
    </li>
    <li>
       
      <b>
        Budget:
      </b>
       
      123456789
       
    </li>
    <li>
       
      <b>
        Only Adult:
      </b>
       
      No
       
    </li>
    <li>
       
      <b>
        Release Date:
      </b>
       
      17-2-2019
       
    </li>
  </ul>
</section>
`;
